### DISCUSSION TITLE
How to implement the WebSocket connection in Dioxus?

### DISCUSSION THREAD

**Discussion by callmeitachi** [DioxusLabs/dioxus#3242] [Q&A]
.

**Comment by brianmay** [comment]
I think you have to do it yourself basically.

Dioxus does not do anything to help you. Which means you have to use the instructions here to get access to the Axum Router: https://docs.rs/dioxus-fullstack/latest/dioxus_fullstack/#axum-integration

There is an example websocket client here: https://github.com/DioxusLabs/dioxus/discussions/1721

And I wrote an example full stack app that does both server and client here(*):

https://github.com/brianmay/dioxus-fs-demo/blob/main/src/main.rs

(*) still under development but does work and doesn't yet support reconnecting if socket goes down.

**Comment by brianmay** [comment]
Also see this response here: https://github.com/DioxusLabs/dioxus/discussions/3378#discussioncomment-11640227
